CRM/MembershipExtras/Job/UpdateMembershipPeriod.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+class CRM_MembershipExtras_Job_UpdateMembershipPeriod {
+
+  /**
+   * @var array
+   */
+  private $overdueContributionStatusIDs = [];
+  
+  /**
+   * Class constructor.
+   */
+  public function __construct() {
+    $this->setOverdueContributionStatusIDs();
+  }
+
+  private function setOverdueContributionStatusIDs() {
+    $contributionStatusesNameMap = CRM_MembershipExtras_Service_ContributionUtilities::getContributionStatusesNameMap();
+    $pendingStatusID = $contributionStatusesNameMap['Pending'];
+    $progressStatusID = $contributionStatusesNameMap['In Progress'];
+    $partialStatusID = $contributionStatusesNameMap['Partially paid'];
+    $overdueStatusID = $contributionStatusesNameMap['Overdue'];
+
+    $this->overdueContributionStatusIDs = [$pendingStatusID, $progressStatusID, $overdueStatusID, $partialStatusID];
+  }
+
+  /**
+   * Starts the scheduled job for disabling overdue membership
+   * periods
+   * 
+   * @return True
+   * 
+   * @throws \Exception
+   */
+  public function run() {
+    $membershipPeriodDao = $this->getMembershipsPeriodsWithOverduePayment();
+    
+    while ($membershipPeriodDao->fetch()) {
+      try {
+        $this->disableMembershipPeriod($membershipPeriodDao->membership_period_id);
+      } catch (Exception $e) {
+        $message = "An error occurred disabling an overdue membership period with id({$membershipPeriodDao->membership_period_id}): " . $e->getMessage();
+
+        throw new Exception($message);
+      }
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Gets membership period IDs with an OVERDUE PAYMENT. The following conditions
+   * must be satisfied:
+   * 
+   * 1- Associated contribution's receive_date is less than (today - $daysToDisableMP)
```

Comment:
no need to describe the two conditions in the docblock, the can be known from the code.

Reply:
Fixed